pkg/controller/node/nodecontroller.go

```diff
@@ -657,6 +741,10 @@ func (nc *NodeController) monitorNodeStatus() error {
 			continue
 		}
 		node := nodeCopy.(*v1.Node)
+
+		// taint or clear NoSchedule taint based on condition change, we need to do this before tryUpdateNodeStatus()
+		nc.addOrRemoveNoScheduleTaintByCondition(node)
```

Comment:
shouldn't we be checking that if node.Spec.Unschedulable is set by cluster admin/operator, then we should not bother about calling addOrRemoveNoScheduleTaintByCondition?